chart/templates/openshift/role.yaml

```diff
@@ -0,0 +1,43 @@
+{{ if .Values.openshift.enabled }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  annotations:
+    "helm.sh/hook": pre-install, pre-upgrade, post-delete
+    "helm.sh/hook-weight": "-9"
```

Comment:
I added the weight because openshift resources (role, SCC, rolebinding ) must be created before anything else.
This would normally happen, however there's a  `kube-prometheus-admission-create` hook, that's why I had to make the resources hooks as well.  The weight is there to tell helm to install openshift resources before `kube-prometheus-admission-create`. I just chose value of `-9` arbitrarily. It is the same for all three resources, because helm creates them in correct order by default, see [hook release lifecycle](https://helm.sh/docs/topics/charts_hooks/#hooks-and-the-release-lifecycle)
  and  [helm install order](https://helm.sh/docs/intro/using_helm/#helm-install-installing-a-package).


I tested it without hook weight today and it works just fine in our openshift cluster, so I removed it for now.

I am not sure about in what order the SCC installs, since it's a custom resource. Maybe further discussion is needed before we merge this PR.

Reply:
AFAIK, when you install the Helm chart, Helm automatically sorts the resources in a specific way to avoid conflicts from the Kubernetes API. This is done by ordering the resources based on a predefined order, hard-coded in the Helm code [here](https://github.com/helm/helm/blob/39ca699ca790e02ba36753dec6ba4177cc68d417/pkg/releaseutil/kind_sorter.go#L31-L66). All resource types that are not in the predefined order will be placed at the end of the installation order.